### DISCUSSION TITLE
jpeg_save with white background?

### DISCUSSION THREAD

**Discussion by 9mm** [libvips/libvips#2242] [Q&A]
I'm using [govips](https://github.com/davidbyttow/govips) to simply convert PNG -> JPG with 100% quality/no color information loss.

Right now I have it working except I would like a white background for transparency instead of black.

I found this thread: https://github.com/libvips/libvips/issues/1449

However I can't find any `Background` option that he references.

Here is the Go version of the code

```go
inputImage, err := vips.NewImageFromFile(filePath)
ep := vips.NewJpegExportParams() 
ep.Quality = 100                 
imageBytes, _, err := inputImage.ExportJpeg(ep)
```

Here is the part of the library of what options I think I can choose from?

https://github.com/davidbyttow/govips/blob/4fdfdb864573ac60097531b46adbb02380f175b2/vips/image.go#L47

```
type ExportParams struct {
	Format             ImageType
	Quality            int
	Compression        int
	Interlaced         bool
	Lossless           bool
	Effort             int
	StripMetadata      bool
	OptimizeCoding     bool          // jpeg param
	SubsampleMode      SubsampleMode // jpeg param
	TrellisQuant       bool          // jpeg param
	OvershootDeringing bool          // jpeg param
	OptimizeScans      bool          // jpeg param
	QuantTable         int           // jpeg param
}
```

**Comment by jcupitt** [comment]
Hi @9mm, it looks like a feature that govips needs to add. In pyvips, for example, you can write:

```
$ python3
Python 3.8.6 (default, Jan 27 2021, 15:42:20) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvips
>>> x = pyvips.Image.new_from_file("PNG_transparency_demonstration.png")
>>> x.write_to_file("x.jpg", background=255)
>>> 
```

To make:

![x](https://user-images.githubusercontent.com/580843/117360077-3fa22480-aeb0-11eb-8d3e-235942b47eaa.jpg)

**Reply by 9mm** [reply]
Hmm I see.... Ok thank you.

**Reply by 9mm** [reply]
Ok -- that is a super helpful reply. I decided to just go with the command line version of vips.

So, if I want to achieve all of the following:

* convert PNG -> JPG
* replace transparency with white pixels
* Make sure to preserve 100% quality (or as much quality as I possibly can given that JPG is lossy)
* Make sure to preserve all weird color settings and color profiles that might affect the color of final image
* Transform image in place (no copy)

This is what I had so far... does this look correct?

   vips copy 800x800.png 'output.jpg[Q=100,background=255]'

The one part i'm not sure on is editing in place, what to use instead of `copy`

**Comment by jcupitt** [comment]
Yes, looks OK. You can do `vips jpegsave` at the command-line to see all the options.

**Comment by matobi** [comment]
Hi, I had the same question, and found that you can call Flatten before exporting (if you don't want to use the command line alternative).

```
err := inputImage.Flatten(&vips.Color{R: 255, G: 255, B: 255})
imageBytes, _, err := inputImage.ExportJpeg(ep)
```
